src/app/portlets/content-types/content-types-resolver.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+import { Resolve, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+import { CrudService } from '../../api/services/crud';
+import { ContentType, CONTENT_TYPE_INITIAL_DATA } from './shared/content-type.model';
+import { ContentTypesInfoService } from '../../api/services/content-types-info';
+import { LoginService } from 'dotcms-js/dotcms-js';
+import { DotRouterService } from '../../api/services/dot-router-service';
+
+@Injectable()
+export class ContentTypeResolver implements Resolve<ContentType> {
+    constructor(
+        private contentTypesInfoService: ContentTypesInfoService,
+        private crudService: CrudService,
+        private loginService: LoginService,
+        private dotRouterService: DotRouterService
+    ) {}
+
+    resolve(route: ActivatedRouteSnapshot): Observable<ContentType> {
+        if (route.paramMap.get('id')) {
+            return this.crudService
```

Comment:
maybe we might change this for a private method something like getContentType

Reply:
Yup, great this way we can have a clear code.